src/LayerPlan.cpp

```diff
@@ -344,18 +345,16 @@ GCodePath& LayerPlan::addTravel(Point p, bool always_retract)
         }
     }
     
-    if (!combed && last_planned_position && !is_first_travel_of_layer) {
-        // no combing? always retract!
-        if (always_retract || !shorterThen(*last_planned_position - p, retraction_config.retraction_min_travel_distance))
-        {
-            if (was_inside) // when the previous location was from printing something which is considered inside (not support or prime tower etc)
-            {               // then move inside the printed part, so that we don't ooze on the outer wall while retraction, but on the inside of the print.
-                assert (extr != nullptr);
-                moveInsideCombBoundary(extr->getSettingInMicrons((extr->getSettingAsCount("wall_line_count") > 1) ? "wall_line_width_x" : "wall_line_width_0") * 1);
-            }
-            path->retract = true;
-            path->perform_z_hop = perform_z_hops;
+    // no combing? retract only when path is not shorter than minimum travel distance
+    if (!combed && !is_first_travel_of_layer && !shorterThen(*last_planned_position - p, retraction_config.retraction_min_travel_distance))
```

Comment:
But the test does look at is_first_travel_of_layer which is the result of testing that last_planned_position has a value.